test/shared/src/test/scala/zio/test/mock/RandomSpec.scala

```diff
@@ -1,363 +1,84 @@
-    SAssert(nextStringWithEmptyData, "MockRandom returns next string when data is empty")
-    SAssert(nextStringWithDefault, "MockRandom returns next string when data is default")
-    SAssert(nextStringWithSequence, "MockRandom returns next string when data is sequence")
-    SAssert(nextStringWithFeed, "MockRandom returns next string when data is feed")
-    SAssert(nextStringWithClear, "MockRandom returns next string when data is clear")
-    SAssert(nextStringWithLength, "MockRandom returns next string when data is single value - respect length")
-    SAssert(
-      nextStringLengthIsOver,
-      "MockRandom returns next string when data is single value - length > length of the next string"
-    )
-
-    SAssert(nextBytesWithSingleValue, "MockRandom returns next bytes when data is single value")
-    SAssert(nextBytesWithEmptyData, "MockRandom returns next bytes when data is empty")
-    SAssert(nextBytesWithDefault, "MockRandom returns next bytes when data is default")
-    SAssert(nextBytesWithSequence, "MockRandom returns next bytes when data is sequence")
-    SAssert(nextBytesWithFeed, "MockRandom returns next bytes when data is feed")
-    SAssert(nextBytesWithClear, "MockRandom returns next bytes when data is clear")
-    SAssert(nextBytesWithLength, "MockRandom returns next bytes when data is single value - length < number of bytes")
-    SAssert(
-      nextBytesLengthIsOver,
-      "MockRandom returns next bytes when data is single value - length > length of the next array"
-    )
-
-    SAssert(
-      shuffleListWithNoElements,
-      "MockRandom shuffle returns the same list in the same order always when list contains no elements"
-    )
-    SAssert(
-      shuffleListWithSingleElement,
-      "MockRandom shuffle returns the same list in the same order always when list contains single element"
-    )
-    SAssert(
-      shuffleListWithTwoElementsSame,
-      "MockRandom shuffle returns the same list in the same order sometimes when list contains two elements"
-    )
-    SAssert(
-      shuffleListWithThreeElementsSame,
-      "MockRandom shuffle returns the same list in the same order sometimes when list contains three elements"
-    )
-    SAssert(
-      shuffleListWithManyElementsSame,
-      "MockRandom shuffle returns the same list in the same order sometimes when list contains many elements"
-    )
-
-    SAssert(
-      shuffleListWithTwoElementsReversed,
-      "MockRandom shuffle returns the same list in different order sometimes when list contains two elements"
-    )
-    SAssert(
-      shuffleListWithThreeElementsReversed,
-      "MockRandom shuffle returns the same list in different order sometimes when list contains three elements"
-    )
-    SAssert(
-      shuffleListWithManyElementsReversed,
-      "MockRandom shuffle returns the same list in different order sometimes when list contains many elements"
-    )
+    SAssert(referentiallyTransparent, "MockRandom referential transparency")
```

Comment:
Really great work! At some point we may want to assert the "bounded" variants generate values within the bounds.

Reply:
Yes I will add a test for this.